app/models/manageiq/providers/azure/manager_mixin.rb

```diff
@@ -10,19 +10,9 @@ def connect(options = {})
   end
 
   def verify_credentials(_auth_type = nil, options = {})
-    require 'azure-armrest'
-    conf = connect(options)
-  rescue ArgumentError => err
-    raise MiqException::MiqInvalidCredentialsError, _("Incorrect credentials - #{err.message}")
-  rescue ::Azure::Armrest::UnauthorizedException, ::Azure::Armrest::BadRequestException
-    raise MiqException::MiqInvalidCredentialsError, _("Incorrect credentials - check your Azure Client ID and Client Key")
-  rescue MiqException::MiqInvalidCredentialsError
-    raise # Raise before falling into catch-all block below
-  rescue => err
-    _log.error("Error Class=#{err.class.name}, Message=#{err.message}, Backtrace=#{err.backtrace}")
-    raise MiqException::MiqInvalidCredentialsError, _("Unexpected response returned from system: #{err.message}")
-  else
-    conf
+    connection_rescue_block do
```

Comment:
Do you need to call `connection_rescue_block` here?

It looks like `connect` will call `raw_connect`, which already wraps the call to `Configuration.new` with `connection_rescue_block`.

Reply:
that's correct, thanks!